autoload/sj/elixir.vim

```diff
@@ -189,3 +189,85 @@ function! sj#elixir#JoinArray()
 
   return 1
 endfunction
+
+function! sj#elixir#SplitPipe()
+  let line = getline('.')
+
+  if line =~ '^\s*|>\s\+'
+    return 0
+  endif
```

Comment:
Here's one example I have laying around that doesn't work well, it seems:

``` elixir
if Enum.member?(unquote(schema).__schema__(:fields), :deleted_at),
  do: query |> where([schema], is_nil(schema.deleted_at))
  else: query
```

Splitting with the cursor on the `where` gives me:

``` elixir
if Enum.member?(unquote(schema).__schema__(:fields), :deleted_at),
  [schema]
  |> do: query |> where(is_nil(schema.deleted_at))
  else: query
```

It could be argued that people should just not attempt this, but I try to add safeguards if I can. Do you think it makes sense to add another check here like:

``` vim
  if line =~ '^\s*\k\+:'
    " Starts with a `keyword:`, part of a dictionary
    return 0
  endif
```

Comment optional, and you could consider joining the pattern to the one above, too.

Reply:
Right, good catch. I didn't think much about single-line pipelines, although maybe there's a case for supporting them somehow.

I guess it would be ambiguous how I'd want to split this:

``` elixir
String.length("foo")
```

I.e. should it be:

``` elixir
"foo"
|> String.length()
```

Or this:

``` elixir
"foo" |> String.length()
```

But at least the joining case is less ambiguous. It seems reasonable that this:

``` elixir
IO.inspect("foo" |> String.length())
```

Could join to this:

``` elixir
IO.inspect(String.length("foo"))
```

And in fact split to this, depending on cursor position:

``` elixir
"foo" |> String.length()
|> IO.inspect()
```

In any case, I think for now I'll have a think about how we can ignore these situations as I reckon it'll probably be a bit of a can of worms that I'm not keen to open now. (But might after getting some use out of these changes and seeing whether I miss the above functionality in the real world...)